### PR TITLE
Dp-1104 update service settings in docker compose

### PR DESCRIPTION
**Pull request by mariana-dias-alves**
## Proposed Changes

  - [x] Updated haproxy
  - [x] Updated redis
  - [x] Update backend (need help of @PedroCristovao-Movai and @diasnad )
  - [x] Update spawner
  - [x] Update ros-master
  - [x] Update ros-tools
  - [x] Update simulator
  
  - [ ] Change repository from nexus to github
  

Updated backend version to backend-noetic:2.4.1.35 and movai-ide-ce to 3.1.0-3 and the IDE flows are not loading. Logs from the backend container were not conclusive. Equivalent versions of frontend packages specified in e.e. manifest, namely, movai-launcher=2.4.0-4 and movai-ide=2.4.0-17 are not available in ppa-public.

Last pr run (successful) : https://github.com/MOV-AI/movai-flow/actions/runs/5682658942/job/15401544856

  
---

- [x] Make sure you are opening from a **topic/feature/bugfix branch**
- [x] Ensure that the PR title represents the desired changes
- [x] Ensure that the PR description detail the desired changes
- [x] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

[^note]:
    Put an `x` into the [ ] to show you have filled the information.
    The template comes from https://github.com/MOV-AI/.github/blob/master/.github/pull_request_template.md
    You can override it by creating .github/pull_request_template.md  in your own repository

### CODE REVIEW

File: docker-compose-host.yml [hunk 1]
```diff
@@ -141,7 +141,7 @@ services:
 
   backend:
     <<: *movai-common
-    image: ${MOVAI_REGISTRY:-pubregistry.aws.cloud.mov.ai}/ce/backend-noetic:1.0.0.23
+    image: ${MOVAI_REGISTRY:-pubregistry.aws.cloud.mov.ai}/ce/backend-noetic:2.4.1.35
```
Comment (AlexFernandes-MOVAI) [inline docker-compose-host.yml:144]: please try and use ghcr here instead of pubregistry

File: docker-compose-host.yml [hunk 2]
```diff
@@ -193,15 +193,15 @@ services:
...
+    image: ${MOVAI_REGISTRY:-pubregistry.aws.cloud.mov.ai}/ce/flow-initiator-noetic:2.4.1.36
     privileged: true
     network_mode: host
     depends_on:
       <<: *movai-common-depend
       backend:
         condition: service_healthy
     healthcheck:
-      test: pgrep -f "/usr/bin/python3 -m flow_initiator"
+      test: pgrep -f "/usr/bin/python3 /usr/local/bin/flow_initiator"
```
Comment (Mograbi) [inline docker-compose-host.yml:204]: flow_initiator is a python module and not a script
it should remain -m flow_initiator

File: .github/workflows/TestOnPR.yml [hunk 3]
```diff
@@ -78,13 +78,13 @@ jobs:
...
         export token_url="$base_url/token-auth/"
-        export authenticated_url="$base_url/api/v1/User/movai/?"
+        export authenticated_url="$base_url/api/v2/User/effective-permissions"
 
         # retry for 1 minute
         count=10
         set +e
         for i in $(seq $count); do
-          status_code=$(curl -k -s -o /dev/null --cookie-jar cookies --header 'Content-Type: application/json' -X POST --data '{"username":"movai","password":"movai","remember":true}' -w "%{http_code}" $token_url)
+          status_code=$(curl -k -s -o /dev/null --header 'Content-Type: application/json' -X POST --data '{"username":"movai","password":"www.mov.ai","remember":true,"domain":"internal"}' -w "%{http_code}" $token_url)
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/TestOnPR.yml]: store username and password in a var, not to edit multiple times

File: docker-compose.yml [hunk 4]
```diff
@@ -194,7 +194,7 @@ services:
       backend:
         condition: service_healthy
     healthcheck:
-      test: pgrep -f "/usr/bin/python3 -m flow_initiator"
+      test: pgrep -f "/usr/bin/python3 /usr/local/bin/flow_initiator"
```
Comment (Mograbi) [inline docker-compose.yml:197]: same as before

File: docker-compose-host.yml (hunk 1 above)
Comment (AlexFernandes-MOVAI) [reply docker-compose-host.yml:144]: as agreed, it will be done in another PR

File: scripts/movai-flow-launch [hunk 5]
```diff
@@ -141,7 +141,7 @@ while [[ -z $(FilterRunningService spawner) ]]; do
   sleep 5
 done
 
-while [[ -z $( $DOCKER_COMPOSE_BIN -f "$compose_conf" exec -u movai spawner bash -c 'pgrep -f "/usr/bin/python3 -m flow_initiator"') ]]
+while [[ -z $( $DOCKER_COMPOSE_BIN -f "$compose_conf" exec -u movai spawner bash -c 'pgrep -f "/usr/bin/python3 /usr/local/bin/flow_initiator"') ]]
```
Comment (Mograbi) [inline scripts/movai-flow-launch:144]: same as before

**Review by AlexFernandes-MOVAI — CHANGES_REQUESTED** [review]

File: docker-compose-host.yml (hunk 2 above)
Comment (mariana-dias-alves) [reply docker-compose-host.yml:204]: The previous command did not work to launch the web app, so I verified the way the app was opened inside the container and it is using this new command

Comment (mariana-dias-alves) [reply docker-compose-host.yml:204]: Screenshot from 2023-07-27 14-58-59

File: .github/workflows/TestOnPR.yml [hunk 6]
```diff
@@ -7,6 +7,9 @@ on:
       - 'releases/**'
     paths-ignore:
       - 'debian/changelog'
+env:
+  username: "movai"
+  password: "www.mov.ai"
```
Comment (AlexFernandes-MOVAI) [inline .github/workflows/TestOnPR.yml]: good practice is to use GitHub secret of the repo to store password

File: docker-compose-host.yml (hunk 2 above)
Comment (AlexFernandes-MOVAI) [reply docker-compose-host.yml:204]: @Mograbi  mariana is right, the health-check is based on what the docker entrypoint currently run

File: docker-compose.yml (hunk 4 above)
Comment (AlexFernandes-MOVAI) [reply docker-compose.yml:197]: docker entrypoint is set like this

**Review by AlexFernandes-MOVAI — CHANGES_REQUESTED** [review]
use github repo secret

**Review by AlexFernandes-MOVAI — APPROVED** [review]

**Pull request merged**